cipher/polybius/polybius.go

```diff
@@ -8,6 +8,54 @@ import (
 	"strings"
 )
 
+type ErrNegativeSize struct{ size int }
```

Comment:
Is there a reason why do you need context aware errors? I don't see any use of context that you're passing - a simple `fmt.Errorf("whatever")` works better (IMO)

You can simply use the `fmt.Errorf("size %d provided", size)` to add context for logging right? This pattern is unusual IMO - unless you want the user to care about different types of `error`s (bad idea in my opinion generally speaking unless you have a custom logging library or custom recovery scenarios - which is not the case here...). I think in this case `error`s should be just `error`s rather than types that satisfy the `error` interface.

The reason being that

```Go

func something() error {
    return yourCustomErr{1}
}

err := something()
customErr, _ := err.(yourCustomErr)
customErr = yourCustomErr{2} // its modifiable 😱 
```

Since its modifiable, then there is no real benefit here (IMO).

I would rather use this pattern instead (but still advise against it in this case)

```Go

type Error string

func (e Error) Error() string {
    return string(e)
}

// added benefit - its immutable but provides no value
// without `fmt.Errorf("%w: size: %d", ErrSizeTooSmall, size)`
// the returned error can be checked like this `if errors.Is(err, polybius.ErrSizeTooSmall)`
const ErrSizeTooSmall = Error("provide size is too small")


func something(size) error {
    if size < 0 {
        return fmt.Errorf("%w: size %d", ErrSizeTooSmall, size) 
    }
}
```

PS: I used to believe that this was the valid and correct pattern but I don't anymore 😓

Reply:
I understand your point of view, I'm checking to change the errors to follow the pattern you suggested me in the last example. The original idea for me was that I added these new types to use them in the errors.As to check the error types returned by the algorithm.

Reply:
Hi @tjgurwara99, I changed the code following your suggestions of removing the context aware errors and use just `fmt.Errorf` and passed the tests again. Let me know if you have any other feedbacks and I will address them. Thanks a lot!